### DISCUSSION TITLE
Download result of custom script as text/csv

### DISCUSSION THREAD

**Discussion by msky17** [netbox-community/netbox#11974] [Help Wanted!]
Hi!
Please help me with some kind of netbox magic.
I have custom sctipt that make some routines work with my ipam and i want to return result of this script as txt or csv file to user.
But with django HttpResponse code 
```       file_location = 'Some_path_to_file'
        try:
           with open(file_location, 'r') as f:
              file_data = f.read()
           response = HttpResponse(file_data, content_type='text/plain')
           response['Content-Disposition'] = 'attachment; filename="filename.txt"'
        return response 
```

i've only got 
<HttpResponse status_code=200, "text/plain">
in output. 
Is it possible to do what I'm trying to do?

**Comment by candlerb** [comment]
It's not well documented, but there's an example custom script which returns CSV data [here](https://docs.netbox.dev/en/stable/customization/custom-scripts/#example).

It looks like your script's `run()` method must return a string containing the actual content that you want the user to download (i.e. the actual CSV data).  But I can't see any way to set content-type or content-disposition on that.  I've not tested it, but I think it's not a downloadable attachment, just a `<pre>` field in the results web page where it can be copy-pasted.

References to the relevant Netbox code: 
* https://github.com/netbox-community/netbox/blob/v3.4.6/netbox/extras/scripts.py#L464
* https://github.com/netbox-community/netbox/blob/v3.4.6/netbox/templates/extras/htmx/script_result.html#L50-L51

**Comment by kkthxbye-code** [comment]
I know that some users save the file to the media folder and then just link to it in the script result. Might be an option.

**Reply by candlerb** [reply]
Good idea. The log messages from scripts and reports can contain Markdown hyperlinks, like `[this](/media/foo/bar.png)`
